src/support/generation.c

```diff
@@ -221,18 +217,12 @@ __gen_oldest(WT_SESSION_IMPL *session, int which)
     WT_CONNECTION_IMPL *conn;
     WT_SESSION_IMPL *s;
     uint64_t oldest, v;
-    uint32_t i, session_cnt;
 
     conn = S2C(session);
 
-    /*
-     * No lock is required because the session array is fixed size, but it may contain inactive
-     * entries. We must review any active session, so insert a read barrier after reading the active
-     * session count. That way, no matter what sessions come or go, we'll check the slots for all of
-     * the sessions that could have been active when we started our check.
-     */
-    WT_ORDERED_READ(session_cnt, conn->session_cnt);
-    for (oldest = conn->generations[which], s = conn->sessions, i = 0; i < session_cnt; ++s, ++i) {
+    oldest = conn->generations[which];
+    WT_CONNECTION_FOREACH_SESSION(s, conn)
```

Comment:
This is a functional change but the session count does seem independent from the oldest generation, in theory the only affect is that we could read an older generation. Would you agree?

Reply:
Yep agreed that this looks fine.
@quchenhao pinging you since we were recently discussing sequential consistency and this code as part of PM-3220.